iodata/formats/wfx.py

```diff
@@ -284,3 +216,148 @@ def _check_tag(f_content: str):
+            for ifn in range(ncart)]
+        for irep in range(ncon):
+            for i, primitive_name in enumerate(batch_primitive_names):
+                ifn = CONVENTIONS[(angmom, 'c')].index(primitive_name)
+                permutation[ibasis + irep * ncart + ifn] = ibasis + irep + i * ncon
+        # WFN uses non-normalized primitives, which will be corrected for
+        # when processing the MO coefficients. Normalized primitives will
+        # be used here. No attempt is made here to reconstruct the contraction.
+        for exponent in batch_exponents:
+            shells.append(Shell(icenter, [angmom], ['c'], np.array([exponent]),
+                                np.array([[1.0]])))
+        # Move on to the next contraction
+        ibasis += ncart * ncon
+    obasis = MolecularBasis(shells, CONVENTIONS, 'L2')
+    assert obasis.nbasis == nbasis
+    return obasis, permutation
+
+
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    data = load_data_wfx(lit)
+
+    # Build the basis set and the permutation needed to regroup shells.
+    obasis, permutation = build_obasis(data['icenters'], data['type'], data['exponents'])
+    # Re-order the mo coefficients.
+    mo_coefficients = data['mo_coeff'][permutation]
+    # Get the normalization of the un-normalized Cartesian basis functions.
+    # Use these to rescale the mo_coefficients.
+    scales = []
+    for shell in obasis.shells:
+        angmom = shell.angmoms[0]
+        for name in obasis.conventions[(angmom, 'c')]:
+            if name == '1':
+                nx, ny, nz = 0, 0, 0
+            else:
+                nx = name.count('x')
+                ny = name.count('y')
+                nz = name.count('z')
+            scales.append(gob_cart_normalization(shell.exponents[0], np.array([nx, ny, nz])))
+    scales = np.array(scales)
+    mo_coefficients /= scales.reshape(-1, 1)
+    mo_count = np.arange(data['mo_coeff'].shape[1])
+    norb = mo_coefficients.shape[1]
+    # make the wavefunction
+    if data['mo_occ'].max() > 1.0:
+        # closed-shell system
+        mo = MolecularOrbitals(
+            'restricted', norb, norb,
+            data['mo_occ'], data['mo_coeff'], data['mo_energy'], None)
+    else:
+        # open-shell system
+        # counting the number of alpha orbitals
+        norba = 1
+        while (norba < mo_coefficients.shape[1]
+               and data['mo_energy'][norba] >= data['mo_energy'][norba - 1]
+               and mo_count[norba] == mo_count[norba - 1] + 1):
+            norba += 1
+        mo = MolecularOrbitals(
+            'unrestricted', norba, norb - norba,
+            data['mo_occ'], data['mo_coeff'], data['mo_energy'], None)
+
+    result = {
+        'title': data['title'],
+        'atcoords': data['atcoords'],
+        'atnums': data['atnums'],
+        'obasis': obasis,
+        'mo': mo,
+        'energy': data['energy'],
+    }
```

Comment:
It would be nice to add `'atgradient': data['gradient']`. There may be other useful additions.

Reply:
Sure, I will do.

Reply:
Fixed.